### ISSUE TITLE
false positive: Lint/ReturnInVoidContext

### ISSUE THREAD

**Issue by dennisvandehoef** [rubocop-hq/rubocop#5330]
It should be possible to do `return super` to save some lines. Super should also not have a return value, so `return super` and `return` have the same return value.

```ruby
def snippets=(snippets)
      snippets = snippets.reject(&:blank?)
      return super if snippets.first.is_a? Snippet
      snippets = snippets.map do |snippet_id|
        Snippet.find snippet_id if snippet_id.to_i > 0
      end
      super
    end
```

version:
```
rubocop -v
0.52.0
```

**Comment by mikegee** [comment]
Unexpectedly, `super` in that subclass's method does provide whatever the superclass's method `return`s (whether the superclass return value is explicit or implicit).

I agree that is would be nice if this claim were true: "Super should also not have a return", I don't believe that Ruby behaves like we prefer.

I tested Ruby 2.2.8.

**Comment by dennisvandehoef** [comment]
Mike, I understand what you say. I mean it more within the scope of my project.
In my opinion, Rubocop should not force me to correct this if it was already bad in the inherited method

Since Rubocop should already check the method called by `super` if it is in my project, it should not return a value. And if it is not a method defined inside my project, I might not be willing to change the way the output of the overwritten version.

**Comment by Drenmi** [comment]
> Since Rubocop should already check the method called by super if it is in my project, it should not return a value.

Unfortunately we can not do cross-references in RuboCop. We look at the files one by one, and don't have any notion of whether the parent class method will return something or not.
